runtime/src/snapshot_utils.rs

```diff
@@ -925,72 +1514,116 @@ fn untar_snapshot_in<P: AsRef<Path>>(
     Ok(account_paths_map)
 }
 
-fn verify_snapshot_version_and_folder(
-    snapshot_version: &str,
-    unpacked_snapshots_dir: &Path,
-) -> Result<(SnapshotVersion, SlotSnapshotPaths)> {
-    info!("snapshot version: {}", snapshot_version);
+fn verify_unpacked_snapshots_dir_and_version(
+    unpacked_snapshots_dir_and_version: &UnpackedSnapshotsDirAndVersion,
+) -> Result<(SnapshotVersion, BankSnapshotInfo)> {
+    let snapshot_version_str = unpacked_snapshots_dir_and_version.snapshot_version.trim();
```

Comment:
maybe `.trim()` could be moved here?: https://github.com/solana-labs/solana/pull/18504/files#r674890169

That's to group similar code together.

Reply:
also, sanitizing untrusted input early in the codepath is always better practice. :)

Reply:
Done!